sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -40,7 +40,7 @@ import org.apache.spark.sql.types.{StructField, StructType}
 case class HadoopFsRelation(
     location: FileIndex,
     partitionSchema: StructType,
-    dataSchema: StructType,
+    dataSchema: StructType, // The top-level columns should not be pruned. Please see SPARK-34897.
```

Comment:
Can we put more details?
```
// The top-level columns in `dataSchema` should match the actual physical file schema, otherwise
// the ORC data source may not work with the by-ordinal mode.
```

Reply:
Done.